kairon/shared/data/audit/processor.py

```diff
@@ -0,0 +1,107 @@
+import json
+from typing import Text
+
+from loguru import logger
+from mongoengine import DoesNotExist
+
+from kairon import Utility
+from kairon.exceptions import AppException
+from kairon.shared.data.audit.data_objects import AuditLogData
+from kairon.shared.data.constant import AuditlogActions
+from kairon.shared.data.data_objects import EventConfig
+
+
+class AuditDataProcessor:
+
+    @staticmethod
+    def log(entity, account: int = None, bot: Text = None, email: Text = None, data: dict = None, **kwargs):
+        """
+        Logs information into auditlog
+
+        :param entity: Type of UserActivity
+        :param account: account id
+        :param bot: bot id
+        :param email: email
+        :param data: dictionary containing data
+        """
+
+        from kairon.shared.account.processor import AccountProcessor
+
+        action = kwargs.get("action")
+        attribute = [{'key': 'bot', 'value': bot}, {'key': 'account', 'value': account}]
```

Comment:
use get_attributes()
account value will be stored as None if it is not provided
pass dict {"bot": bot, "account": account} to get_attributes() and it will take care of this scenario